corehq/apps/case_search/filter_dsl.py

```diff
@@ -213,6 +213,13 @@ def _comparison(node):
             )
 
     def visit(node):
+
+        if isinstance(node, FunctionCall):
+            if node.name in XPATH_FUNCTIONS:
+                return XPATH_FUNCTIONS[node.name](node, fuzzy)
```

Comment:
We can't support all functions here, only 'query' function: `date('2022-01-01')` is not a valid query.

We should probably separate value functions from query functions since the new functions in this PR don't make sense as value functions:
```
# good
dob = date('2019-01-01')

# bad
dob = selected(prop, 'value')
```

Reply:
I ended up doing this in the subcase PR: https://github.com/dimagi/commcare-hq/pull/31115/files#diff-466c8903c1e7201de8bf4c76a3abd642f389e285d8bcf27f4509c834b0aae580R224-R231

Feel free to copy it over, I'll handle any conflicts on my PR.